qcodes/tests/dataset/test_database_creation_and_upgrading.py

```diff
@@ -151,6 +155,76 @@ def test_perform_actual_upgrade_1_to_2():
         assert len(c.fetchall()) == 2
 
 
+def test_perform_actual_upgrade_2_to_3():
+
+    v2fixpath = os.path.join(fixturepath, 'db_files', 'version2')
+
+    if not os.path.exists(v2fixpath):
+        pytest.skip("No db-file fixtures found. You can generate test db-files"
+                    " using the scripts in the legacy_DB_generation folder")
+
+    dbname_old = os.path.join(v2fixpath, 'empty.db')
+
+    with temporarily_copied_DB(dbname_old, debug=False, version=2) as conn:
+
+        assert get_user_version(conn) == 2
+
+        desc_query = 'SELECT run_description FROM runs'
+
+        with pytest.raises(RuntimeError):
+            atomic_transaction(conn, desc_query)
+
+        perform_db_upgrade_2_to_3(conn)
+
+        assert get_user_version(conn) == 3
+
+        c = atomic_transaction(conn, desc_query)
+        assert len(c.fetchall()) == 0
+
+    dbname_old = os.path.join(v2fixpath, 'some_runs.db')
+
+    with temporarily_copied_DB(dbname_old, debug=False, version=2) as conn:
```

Comment:
two tests in one? i'd recommend to split the two - one for an empty db, and the second for "some_runs" db.

Reply:
Done.